Products/zms/_confmanager.py

```diff
@@ -1096,11 +1096,15 @@ def getCharFormats(self):
 
     def getCatalogAdapter(self):
       from Products.zms import IZMSCatalogAdapter, ZMSZCatalogAdapter
-      for ob in self.objectValues():
-        if IZMSCatalogAdapter.IZMSCatalogAdapter in list(providedBy(ob)):
-          return ob
+      path_nodes = self.breadcrumbs_obj_path()

```

Comment:
This will traverse the whole hierarchy. Catalog-adapters can only reside in ZMS-instances.

Reply:
We can fix this later with a separate issue.